### ISSUE TITLE
[DEMO] Training QSP Phase Angles via JAX Optimization (PennyLane + JAX + Optax)

### ISSUE THREAD

**Issue by rosspeili** [PennyLaneAI/demos#1730]
#### General information

**Name**
Vladimiros Peilivanidis (Ross Peili)

**Affiliation** (optional)
Independent / ARPA Hellenic Logical Systems (arpacorp.net)

**Twitter** (optional)
@rosspeili 

**LinkedIn**
https://linkedin.com/in/ross-peili

---

#### Demo information

**Title**
Training QSP Phase Angles via JAX Optimization

**Abstract**
This demo shows how to train Quantum Signal Processing (QSP) phase angles from scratch using JAX and Optax, rather than computing them analytically. Starting from random initialization, an Adam optimizer recovers a degree-5 Chebyshev approximation of sin(x) by differentiating through a fully differentiable, flat PennyLane circuit. This approach is distinct and highly powerful, as it allows for optimizing QSP sequences when the target polynomial is implicitly defined by a loss function, when the circuit is embedded in a larger variational routine, or when standard analytic solvers become numerically unstable at high degrees.

**Relevant links**
- Repository: https://github.com/rosspeili/qsp-pennylane-demo
- Notebook: https://github.com/rosspeili/qsp-pennylane-demo/blob/main/demo.ipynb
- Martyn et al., [A Grand Unification of Quantum Algorithms](https://arxiv.org/abs/2105.02859), PRX Quantum 2021
- Gilyen et al., [Quantum singular value transformation](https://arxiv.org/abs/1806.01838), STOC 2019

**Comment by drdren** [comment]
Hi @rosspeili , I really enjoyed reading your demo! It read well and I think it's a topic that folks will find helpful! Just a couple of minor comments: Can you change "import pennylane as qml" to "import pennylane as qp" and therefore qml.ABC must become qp.ABC? This is to reflect our recent change from being more than just quantum machine learning. You can read more about it in this blog post: https://pennylane.ai/blog/2026/03/import-pennylane-as-qp-growing-beyond-just-qml. 

I think the demo is more impactful than your abstract suggests. In particular, the introduction section makes clear why your approach is different to typical approaches. Could you modify the abstract a little to reflect more of this distinction? 

After that, we can move on to making a card on the community demo page and marketing your demo. Are you fine with us tagging your social media links in our LinkedIn, X (Twitter), Slack, Discord advertising? 

cc: @CatalinaAlbornoz

**Comment by rosspeili** [comment]
Hi @CatalinaAlbornoz, @drdren 

Thank you for the kind words and helpful feedback! 

I've just pushed a new commit to the repo that updates all instances of `import pennylane as qml` and `qml.*` to `import pennylane as qp` and `qp.*` to correctly reflect the new naming convention. 

I entirely agree regarding the abstract; I've updated the issue description above to incorporate the details from the introduction to better distinguish this method from typical analytic approaches.

Regarding the social media marketing and the community demo page: Yes, absolutely! Feel free to tag me using my handle `@rosspeili` on X/Twitter and via my LinkedIn profile (https://linkedin.com/in/ross-peili). I'm very excited for this!

Let me know if there's anything I can help with. <3

**Comment by drdren** [comment]
Thanks, @rosspeili ! It looks great! I am sure that people will find this interesting to read. In terms of next steps, I will create a card for your demo at https://pennylane.ai/qml/demos_community with your title and abstract linking to the repo https://github.com/rosspeili/qsp-pennylane-demo next week. We will set up the marketing of your demo next week also (probably Wednesday, the 29th of April) but that is tentative for now. 

Keep an eye out on PennyLane's social media pages and share the demo when it comes out! We will be sure to tag your X/Twitter and your LinkedIn handles. Thanks!

cc: @CatalinaAlbornoz

**Comment by drdren** [comment]
Also, how should we refer to you? Vladimiros Peilivanidis or Ross Peili?

**Comment by rosspeili** [comment]
Appreciate it guys. You can call me Ross or Vlad. ❤️

**Comment by drdren** [comment]
Thanks! Do you have a preference of how we refer to you in our social media posts?

**Comment by rosspeili** [comment]
Just tag, in Twitter / Threads @rosspeili and on LinkedIn should show the full name. If you want persistence across post, probably full name Vladimiros Peilivanidis.

**Comment by CatalinaAlbornoz** [comment]
Thanks @rosspeili !

**Comment by drdren** [comment]
Hi @rosspeili ! Just to let you know, your community demo is now live on the website at https://pennylane.ai/qml/demos_community! I am also confirming that we are set to advertise your demo tomorrow (Wednesday, the 29th of April 2026) on LinkedIn, X/Twitter, Discord, and Slack --- so keep your eyes peeled and feel free to share the links. 

Again, thanks for sending the demo over!

cc: @CatalinaAlbornoz

**Comment by rosspeili** [comment]
That is awesome @drdren @CatalinaAlbornoz thanks a lot. I will wait for the social shares and repost yours instead of fragmenting it. 🙏

**Comment by drdren** [comment]
Thanks @rosspeili. The social media posts are live! You have generated quite a few questions. That's a great sign of contributing something cool for the community. You might want to even start up a PennyLane forum thread just to collect & preserve these. 

Thanks for your demo. Feel free to share the links below! 

LinkedIn: https://www.linkedin.com/feed/update/urn:li:activity:7455225065035616256
X/Twitter: https://x.com/PennyLaneAI/status/2049458976400556192

cc: @CatalinaAlbornoz

**Comment by drdren** [comment]
Hello @rosspeili , just to let you know, other websites have picked up your work!

https://quantumcomputingreport.com/new-open-source-tool-uses-gradient-descent-to-determine-qsp-phase-angles/

**Comment by rosspeili** [comment]
hey @drdren this is amazing! thanks for the heads up <3

**Comment by rosspeili** [comment]
Hey @CatalinaAlbornoz @drdren could please endorse me on arxiv with this code? https://arxiv.org/auth/endorse?x=TO8Z83 so I am able to post a relevant paper on this demo in quantum-ph? thanks in advance. <3

**Comment by CatalinaAlbornoz** [comment]
Hi @rosspeili, unfortunately we cannot do arXiv endorsements.

**Comment by rosspeili** [comment]
Understood and respected 🙏

**Comment by CatalinaAlbornoz** [comment]
Thanks for understanding.

**Comment by rosspeili** [comment]
Hey @CatalinaAlbornoz @drdren I self published the relevant paper, and expanded the demo quite a bit. 
Paper: https://doi.org/10.5281/zenodo.20645402
New repo: https://github.com/rosspeili/trainable-qsp-angles

**Comment by CatalinaAlbornoz** [comment]
Hi @rosspeili ,

That's great to see! I see that you linked the new repo in the readme of your old repo which gives people a good transition. At the moment we don't have the capacity to do a review of the new repo so for now we'll have to keep the old one linked in the community demo.
